### PR TITLE
Update the project structure

### PR DESCRIPTION
**Pull request by stankudrow**
Goals covered:
- groom the `pyproject.toml` file (done with pdm, but the pdm.lock is not included):
  1. specify minimum python version (the py3.9 is chosen according to the EOL)
  2. add optional dependencies: the `lint` section with the ruff linter and `test` with the pytest framework
- lint and format the whole code base with `ruff` 
- refactor the test code base
- enrich the .gitignore file (not a template as it was in the #13 )

### CODE REVIEW

File: leb128/__init__.py [hunk 1]
```diff
@@ -10,14 +10,24 @@
...
-        assert i >= 0
+
+        if i < 0:
+            msg = f"{i} < 0"
+            raise Leb128Error(msg)
+
         r = []
         while True:
-            byte = i & 0x7f
+            byte = i & 0x7F
```
Comment (stankudrow) [inline leb128/__init__.py:30]: ruff

File: leb128/__init__.py [hunk 2]
```diff
@@ -10,14 +10,24 @@
...
+    """Unsigned LEB128 codec."""
+
     @staticmethod
     def encode(i: int) -> bytearray:
         """Encode the int i using unsigned leb128 and return the encoded bytearray."""
-        assert i >= 0
+
+        if i < 0:
+            msg = f"{i} < 0"
+            raise Leb128Error(msg)
```
Comment (stankudrow) [inline leb128/__init__.py:26]: https://stackoverflow.com/questions/13534563/is-it-bad-practice-to-use-assert-to-verify-assumptions-about-arguments-in-python

File: pyproject.toml [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
 name = "leb128"
 version = "1.0.8"
+description = "LEB128 (Little Endian Base 128)"
 authors = [
-  { name="Mohanson", email="mohanson@outlook.com" },
+    {name = "Mohanson", email = "mohanson@outlook.com"},
 ]
-description = "LEB128(Little Endian Base 128)"
+dependencies = []
+requires-python = ">=3.9"
```
Comment (stankudrow) [inline pyproject.toml:9]: https://endoflife.date/python

**Comment by mohanson** [comment]
Hi, thanks for your PR, but I will not accept this modification.

Project structure and code format are very personal things, even the so-called best practices are essentially just the personal preferences of the person who came up with the concept.

**Pull request closed without merging**